src/lsps2/client.rs

```diff
@@ -405,6 +431,18 @@ where
 						action: ErrorAction::IgnoreAndLog(Level::Info),
 					})?;
 
+				// Reject the buy response if we disallow client_trusts_lsp and the LSP requires
+				// it.
+				if !self.config.client_trusts_lsp && result.client_trusts_lsp {
+					peer_state.remove_inbound_channel(jit_channel_id);
+					return Err(LightningError {
+						err: format!(
+							"Aborting JIT channel flow as the LSP requires 'client_trusts_lsp' mode, which we disallow"
+						),
+						action: ErrorAction::IgnoreAndLog(Level::Info),
+					});
+				}
```

Comment:
Hm, there's currently no way for the end-user client to detect this happened, right? Might need to surface some kind of error event or similar to the user so they can surface to wallet user (or retry with different lsp or however they want to handle it).

In mutiny's current implementation this would just end up with a infinite spinner.

Reply:
Fine with this being a follow up though